88.md

```diff
@@ -0,0 +1,146 @@
+  "id": "30efed56a035b2549fcaeec0bf2c1595f9a9b3bb4b1a38abaf8ee9041c4b7d93",
+  "sig": "f2cb581a84ed10e4dc84937bd98e27acac71ab057255f6aa8dfa561808c981fe8870f4a03c1e3666784d82a9c802d3704e174371aa13d63e2aeaf24ff5374d9d"
+}
+```
+
+## DLC Offers
+
+DLCs are done between two parties and Nostr provides a unique opportunity to create a global market for DLCs. This
+market can be constructed with a simple list of DLC offers that can be filtered by the user's wallet. The user can then
+select an offer and send a message to the offerer to begin the DLC process. `kind:30088` is an event that contains a DLC
+offer.
+
+A parameterized replaceable event is used to allow the offerer to update the offer or remove it from the market. The `d`
+tag should be the Offer's id to allow replacement of the offer. If the user wishes to remove the offer from the market,
+they can replace the event with an expires tag that is in the past.
+
+### `kind:30_088`
+
+```jsonc
+{
+  "kind": 30088,
+  "content": "base64 contract details",
+  "tags": [
+    [
+      "relays",
+      "wss://nostr.mutinywallet.com",
+      "wss://relay.damus.io"
+    ],
+    [
+      "expires",
+      "1695327657"
+    ],
+    [
+      "d",
+      "6423394a32f37a4f0590e74f9308f5aee06e59e14f50c4cd6d59365caed13cb7"
+    ],
+  ],
+  "pubkey": "97c70a44366a6535c145b333f973ea86dfdc2d7a99da618c40c64705ad98e322",
+  "created_at": 1679673265,
+  "id": "30efed56a035b2549fcaeec0bf2c1595f9a9b3bb4b1a38abaf8ee9041c4b7d93",
+  "sig": "f2cb581a84ed10e4dc84937bd98e27acac71ab057255f6aa8dfa561808c981fe8870f4a03c1e3666784d82a9c802d3704e174371aa13d63e2aeaf24ff5374d9d"
+}
+```
+
+## DLC Oracle Gossip
+
+DLCs require an oracle to attest to the outcome of real world events. This is done by the oracle signing a message
+containing the outcome of the event. Before they attest to the outcome, they must create an announcement where they
+publish the intent to sign the future event. This announcement is then used by the DLC participants to create the
+contract. Here we define two events, `kind:88` and `kind:89` that are used to publish the oracle's announcement and
+attestations respectively.
+
+### `kind:88`
+
+```jsonc
+{
+  "kind": 88,
+  "content": "base64 oracle annoucement",
+  "tags": [
+    [
+      "relays", // the relays the oracle will publish attestations to
+      "wss://nostr.mutinywallet.com",
+      "wss://relay.damus.io"
+    ],
+  ],
+  "pubkey": "97c70a44366a6535c145b333f973ea86dfdc2d7a99da618c40c64705ad98e322",
+  "created_at": 1679673265,
+  "id": "30efed56a035b2549fcaeec0bf2c1595f9a9b3bb4b1a38abaf8ee9041c4b7d93",
+  "sig": "f2cb581a84ed10e4dc84937bd98e27acac71ab057255f6aa8dfa561808c981fe8870f4a03c1e3666784d82a9c802d3704e174371aa13d63e2aeaf24ff5374d9d"
+}
+```
+
+### `kind:89`
+
+```jsonc
+{
+  "kind": 89,
+  "content": "base64 oracle attestation",
+    "tags": [
+    [
+      "e", // the event id of the announcement
+      "30efed56a035b2549fcaeec0bf2c1595f9a9b3bb4b1a38abaf8ee9041c4b7d93",
+    ],
```

Comment:
[Oracle announcements also contain an `event_id` field](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Messaging.md#oracle_event), and [so do oracle attestations](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Messaging.md#oracle_attestation). Announcement event IDs _must_ be distinct from the Nostr `kind:88` event IDs (due to the fact that the announcement is part of the preimage for the nostr event ID). This might cause some confusion.

Perhaps we should clarify that the `e` tag must be the _Nostr event ID,_ not the _announcement event ID,_ and that the `e` tag should be used to look up the Nostr event which embeds an oracle announcement. 

One might want to enforce on the client that the `oracle_attestation` data embeds the same _announcement event ID_ that you find by looking up the event indicated by the `e` tag. In pseudocode:

```python
def validate_attestation_event(attestation_event):
  oracle_attestation = decode_attestation(attestation_event.content)
  announcement_event = nostr_event_lookup(attestation_event.tags.e)
  oracle_announcement = decode_announcement(announcement_event.content)
  assert oracle_announcement.event_id == oracle_attestation.event_id
```